chore: refactor app list into common component

ty-kerr opened the pull request:
## What does this PR do?

I refactored the App List component used in settings/conferencing and installed/[category], into one component file.

Fixes #7521 

## Type of change

<!-- Please delete bullets that are not relevant. -->

- Chore (refactoring code, technical debt, workflow improvements)


## How should this be tested?

- [ x] Test A : open both app list in settings/my-account/conferencing and apps/installed/conferencing. check if the app list is rendered, try to delete and and set new default. Changes on one should reflect on the other. 
- [ ] Test B

## Mandatory Tasks

- [ x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

Commit:
refactor app list into common component

ty-kerr committed:
Merge branch 'main' into 1182-refactor-app-list-my-settings-conferencing-and-installed-app-conferencing

PeerRich commented:
@ty-kerr would you mind fixing the merge conflict? We will then look into this with higher priority and I'll set a bounty for your extra work! <3 thank you

ty-kerr committed:
merge latest main into branch

ty-kerr committed:
fix add conferencing button href

ty-kerr commented:
> @ty-kerr would you mind fixing the merge conflict? We will then look into this with higher priority and I'll set a bounty for your extra work! <3 thank you

Fixed all merge conflicts, and updated href for AddConferenceButton to "/apps/categories/conferencing" from "/apps/categories/video"

ty-kerr committed:
merge main into branch

sean-brydon commented:
From a code perspective this looks a lot cleaner! Will test it all works as expected after i've ate some breakfast! 

Do you think you can fix eslint warnings?

sean-brydon commented on apps/web/components/apps/App.tsx:
```diff
@@ -186,19 +169,28 @@ const Component = ({
                     }
                     return (
-                      <InstallAppButtonChild appCategories={categories} userAdminTeams={appDbQuery.data?.userAdminTeams} addAppMutationInput={{ type, variant, slug }} multiInstall {...props} />
+                      <Button
+                        StartIcon={Plus}
+                        {...props}
+                        // @TODO: Overriding color and size prevent us from
+                        // having to duplicate InstallAppButton for now.
+                        color="primary"
+                        size="base"
+                        data-testid="install-app-button">
+                        {t("install_another")}
+                      </Button>
```
@joeauyeung @hariombalhara Will this have any effect not using the install app button child? Not something im too familiar with

hariombalhara commented on apps/web/components/apps/App.tsx:
```diff
@@ -1,36 +1,22 @@
-import type { ButtonProps } from "@calcom/ui";
-import type { AppFrontendPayload as App } from "@calcom/types/App";
 import { Button, showToast, SkeletonButton, SkeletonText, HeadSeo, Badge } from "@calcom/ui";
-import {
-  Dropdown,
-  DropdownMenuTrigger,
-  DropdownMenuContent,
-  DropdownMenuPortal,
-  DropdownMenuLabel,
-  DropdownItem,
-  Avatar,
-} from "@calcom/ui";
```
Looks like merge conflict resolution issue. These are being used.

hariombalhara commented on apps/web/components/apps/App.tsx:
```diff
@@ -77,14 +63,11 @@ const Component = ({
     { appType: type },
     {
-      onSettled(data) {
-        const credentialsCount = data?.credentials.length || 0
-        setShowDisconnectIntegration(data?.userAdminTeams.length ? credentialsCount >= data?.userAdminTeams.length : credentialsCount > 0);
-        setExistingCredentials(data?.credentials.map(credential => credential.id) || []);
```
Needs a comment on why this code was modified.

ty-kerr replied:
i think this was just prettier and eslint doing its thing

ty-kerr replied:
@hariombalhara ive reverted all changes to app.tsx.

hariombalhara reviewed (changes requested):
Thanks for the PR @ty-kerr but not able to test it out.
<img width="1479" alt="Screenshot 2023-07-24 at 4 28 07 PM" src="https://github.com/calcom/cal.com/assets/1780212/7878b010-2bc5-4270-9dd5-9463711f7f15">

ty-kerr commented:
> Thanks for the PR @ty-kerr but not able to test it out. <img alt="Screenshot 2023-07-24 at 4 28 07 PM" width="1479" src="https://user-images.githubusercontent.com/1780212/255569966-7878b010-2bc5-4270-9dd5-9463711f7f15.png">

ill look into it

ty-kerr committed:
merge main

ty-kerr committed:
revert app.tsx

ty-kerr commented:
@hariombalhara the revert of app.tsx seems to fix the error above, please test again.

hariombalhara commented on apps/web/components/apps/App.tsx:
```diff
@@ -492,4 +492,4 @@ const InstallAppButtonChild = ({
     </Dropdown>
   );
-};
+};
```
```suggestion
};

```

hariombalhara committed:
Update apps/web/components/apps/App.tsx

hariombalhara committed:
Merge remote-tracking branch 'origin/main' into 1182-refactor-app-list-my-settings-conferencing-and-installed-app-conferencing

hariombalhara commented:
Resolved a merge conflict @ty-kerr

hariombalhara commented on apps/web/components/apps/AppList.tsx:
```diff
@@ -0,0 +1,231 @@
+        </DropdownItem>
+      </DropdownMenuItem>
+    );
+  }
```
Below code is dead it seems. I couldn't find a case when this would hit.
`credentialId` should be set when the app is installed and `installed` prop is always set to true

@ty-kerr can you confirm this please.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=9754) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=9754) before we can accept your contribution.<br/>**0** out of **3** committers have signed the CLA.<br/><br/>:x: Ty Kerr<br/>:x: ty-kerr<br/>:x: hariombalhara<br/><hr/>**Ty Kerr** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=9754) it.</sub>

ty-kerr committed:
Merge remote-tracking branch 'origin' into 1182-refactor-app-list-my-settings-conferencing-and-installed-app-conferencing

PeerRich committed:
Merge branch 'main' into 1182-refactor-app-list-my-settings-conferencing-and-installed-app-conferencing

hariombalhara reviewed (approved):
Retested.
- Installation of Analytics app for team and individual
- Installing of conferencing apps
- /settings/account/conferencing listing
- /apps/installed -> Disconnect
